pre-commit.yaml

```
@@ -0,0 +1,57 @@
+package:
+  name: pre-commit
+  version: 4.0.1
+  epoch: 0
+  description: "A framework for managing and maintaining multi-language pre-commit hooks"
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - python3
+      - git
+
+environment:
+  contents:
+    repositories:
+      - https://packages.wolfi.dev/os
+    keyring:
+      - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+    packages:
+      - build-base
+      - py3-toml
+      - py3-virtualenv
+      - py3-platformdirs
+      - py3-gpep517
+      - py3-setuptools
+      - py3-wheel
+      - py3-distlib
+      - py3-pytest
+      - git      
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/pre-commit/pre-commit
+      tag: v4.0.1
+      expected-commit: cc4a52241565440ce200666799eef70626457488
+
+  - name: Python Build
+    uses: python/build-wheel
+
+  - name: Python Install
+    uses: python/install
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: pre-commit/pre-commit
+    strip-prefix: v
+
+test:
+  pipeline:
+    - uses: python/import
+    - runs: |
+        import pre_commit
+        from pre_commit.main import main
```

The tests are failing:

```bash
% make test/pre-commit
2025/01/05 19:17:58 INFO running step "python/import"
2025/01/05 19:17:58 WARN + '[' -d /home/build ]
2025/01/05 19:17:58 WARN + cd /home/build
2025/01/05 19:17:58 WARN + exit 0
2025/01/05 19:17:58 WARN + '[' -d /home/build ]
2025/01/05 19:17:58 WARN + cd /home/build
2025/01/05 19:17:58 WARN + set +x
2025/01/05 19:17:58 INFO using python /usr/bin/python3.13
2025/01/05 19:17:58 INFO No package specified for import.
```


```suggestion
test:
  pipeline:
    - uses: python/import
      with:
        import: pre_commit

    - name: Basic command tests
      runs: |
        pre-commit --version | grep -q "${{package.version}}"
        pre-commit --help | grep -q "usage:"
```

Also, tests failing as pre-commit looks to need 'cfgv', which isn't packaged. Tried hacking in 'pip install cfgv' and theres other issues with missing deps. This will need more work and isn't workable in its current state

---

i’ve also packaged [cfgv](https://github.com/asottile/cfgv) and tested it locally. would you recommend pushing the py3-cfgv package before proceeding with testing the pre-commit package import?